core/src/main.ts

```diff
@@ -0,0 +1,150 @@
+import { getListeners } from './listener/api'
+import { postprocessListeners } from './listener/utils'
+import { buildReporter as buildDataFeedReporter } from './reporter/data-feed'
+import { buildReporter as buildL2DataFeedReporter } from './reporter/data-feed-L2'
+import { buildReporter as buildRequestResponseReporter } from './reporter/request-response'
+import { buildReporter as buildL2RequestResponseFulfillReporter } from './reporter/request-response-L2-fulfill'
+import { buildReporter as buildL2RequestResponseRequestReporter } from './reporter/request-response-L2-request'
+import { IReporters } from './reporter/types'
+import { buildReporter as buildVrfReporter } from './reporter/vrf'
+import { buildReporter as buildL2VrfFulfillReporter } from './reporter/vrf-L2-fulfill'
+import { buildReporter as buildL2VrfRequestReporter } from './reporter/vrf-L2-request'
+import { CHAIN, REDIS_HOST, REDIS_PORT } from './settings'
+
+const LISTENERS: IListeners = {
+  DATA_FEED: buildDataFeedListener,
+  VRF: buildVrfListener,
+  REQUEST_RESPONSE: buildRequestResponseListener,
+  DATA_FEED_L2: buildL2DataFeedListener,
+  VRF_L2_REQUEST: buildVrfL2RequestListener,
+  VRF_L2_FULFILL: buildVrfL2FulfillListener,
+  REQUEST_RESPONSE_L2_REQUEST: buildRequestResponseL2RequestListener,
+  REQUEST_RESPONSE_L2_FULFILL: buildRequestResponseL2FulfillListener
+}
+
+const WORKERS: IWorkers = {
+  DATA_FEED: dataFeedWorker,
+  VRF: vrfWorker,
+  REQUEST_RESPONSE: requestResponseWorker,
+  DATA_FEED_L2: l2DataFeedWorker,
+  VRF_L2_REQUEST: l2VrfRequestWorker,
+  VRF_L2_FULFILL: l2VrfFulfillWorker,
+  REQUEST_RESPONSE_L2_REQUEST: l2RequestResponseRequestWorker,
+  REQUEST_RESPONSE_L2_FULFILL: l2RequestResponseFulfillWorker
+}
+
+const REPORTERS: IReporters = {
+  DATA_FEED: buildDataFeedReporter,
+  VRF: buildVrfReporter,
+  REQUEST_RESPONSE: buildRequestResponseReporter,
+  DATA_FEED_L2: buildL2DataFeedReporter,
+  VRF_L2_REQUEST: buildL2VrfRequestReporter,
+  VRF_L2_FULFILL: buildL2VrfFulfillReporter,
+  REQUEST_RESPONSE_L2_REQUEST: buildL2RequestResponseRequestReporter,
+  REQUEST_RESPONSE_L2_FULFILL: buildL2RequestResponseFulfillReporter
+}
+
+const LOGGER = buildLogger()
+const FILE_NAME = import.meta.url
+
+async function startListenerService(service: string, redisClient: RedisClientType) {
+  const listenersRawConfig = await getListeners({ service, chain: CHAIN })
+  const listenersConfig = postprocessListeners({
+    listenersRawConfig,
+    service,
+    chain: CHAIN,
+    logger: LOGGER
+  })
+
+  if (!LISTENERS[service] || !listenersConfig[service]) {
+    LOGGER.error({ name: 'main', file: FILE_NAME, service }, 'service')
+    throw new OraklError(OraklErrorCode.UndefinedListenerRequested)
+  }
+
+  LISTENERS[service](listenersConfig[service], redisClient, LOGGER)
+  LOGGER.info('Listener launched')
+}
+
+function startWorkerService(service: string, redisClient: RedisClientType) {
+  WORKERS[service](redisClient, LOGGER)
```

Comment:
How about adding some checks for worker and reporter too?
```
if (!WORKERS[servce]) {
throw...
}

if (!REPORTERS[service]) {
throw...
}
```

Reply:
or if you think it is unnecessary since it's being already checked from loadArgs, maybe checking `!LISTENERS[service]` from `startListenerService` can be removed